### PR TITLE
[MRG+1] Adding a resample_to_img function

### LINKED ISSUE #1004: Add a resample_to_img
We already have 
```
resample_img(img, target_affine=None, target_shape=None, interpolation='continuous', copy=True, order='F')
```
http://nilearn.github.io/modules/generated/nilearn.image.resample_img.html

It would be nice to have a variant
```
resample_to_img(source_img, target_img, interpolation='continuous', copy=True, order='F')
```
that would take the shape and affine from "target_img".

It's just a simple helper function, but it would make the life of many non advanced users simpler.

We should reference each function in there respective seealso section of the docstrings.

### PR DESCRIPTION
**Pull request by aabadie**
Fix #1004

Maybe some examples in the documentation should added as well.

### CODE REVIEW

File: nilearn/image/resampling.py [hunk 1]
```diff
@@ -306,6 +306,11 @@ def resample_img(img, target_affine=None, target_shape=None,
         input image, resampled to have respectively target_shape and
         target_affine as shape and affine.
 
+    See Also
+    --------
+    nilearn.resampling.resample_to_img : To resample an image on a target
+    image.
```
Comment (GaelVaroquaux) [inline nilearn/image/resampling.py]: Did you check that this renders well. I have doubt on the indentation in seealso for multiple lines.

File: nilearn/image/resampling.py [hunk 2]
```diff
@@ -484,6 +489,59 @@ def resample_img(img, target_affine=None, target_shape=None,
...
+    Returns
+    -------
+    resampled: nibabel.Nifti1Image
+        input image, resampled to have respectively target image shape and
+        affine as shape and affine.
+
+    See Also
+    --------
+    nilearn.resampling.resample_img : To resample an image with using a given
+    shape and affine instead of a target image.
```
Comment (GaelVaroquaux) [inline nilearn/image/resampling.py]: Same comment about seealso.

**Comment by GaelVaroquaux** [comment]
Looks great!

Can you add the function to doc/module/reference.rst, and also in the narrative documentation where resample_img is documented (do a git graph resample_img).

**Comment by aabadie** [comment]
@GaelVaroquaux, I pushed a couple of commits that address your comment. I also linked the example using the resample_img function to the function documentation.

File: examples/04_manipulating_images/plot_affine_transformation.py [hunk 3]
```diff
@@ -11,10 +11,10 @@
 matrix yields (x, y, z, 1), a 4-vector containing the millimeter position of
 the voxel.
 
-The resampling procedure in `resample_img` can attribute a new affine matrix
-and a new shape to your Nifti image while keeping its representation in
-millimeter space exactly the same (up to sampling error and possible
-clipping).
+The resampling procedure in :func:`nilearn.image.resample_img` can attribut
```
Comment (bthirion) [inline examples/04_manipulating_images/plot_affine_transformation.py]: attribute

File: nilearn/image/resampling.py [hunk 4]
```diff
@@ -484,6 +488,58 @@ def resample_img(img, target_affine=None, target_shape=None,
...
+    source_img: Niimg-like object
+        See http://nilearn.github.io/manipulating_visualizing/manipulating_images.html#niimg.
+        Image(s) to resample.
+
+    target_img: Niimg-like object
+        See http://nilearn.github.io/manipulating_visualizing/manipulating_images.html#niimg.
+        Reference image taken for resampling.
+
+    interpolation: str, optional
+        Can be 'continuous' (default) or 'nearest'. Indicate the resample
```
Comment (bthirion) [inline nilearn/image/resampling.py]: Indicates

**Comment by bthirion** [comment]
LGTM otherwise.

**Comment by GaelVaroquaux** [comment]
LGTM aside from the typo spotted by Bertrand

**Comment by aabadie** [comment]
> LGTM aside from the typo spotted by Bertrand
Fixed !

**Comment by aabadie** [comment]
> LGTM aside from the typo spotted by Bertrand
@GaelVaroquaux, fixed it (just in case you missed my previous comment).

File: nilearn/image/resampling.py (hunk 4 above)
Comment (aabadie) [inline nilearn/image/resampling.py]: Done (sorry I missed this comment)

**Comment by GaelVaroquaux** [comment]
Merging. Thank you (you forgot to add in whats_new, and a versionadded in the docstring would have been great, but I'll do that myself).

**Pull request merged**